skxray/correlation.py

```diff
@@ -0,0 +1,319 @@
+
+    References: text [1]_
+
+    .. [1] D. Lumma, L. B. Lurio, S. G. J. Mochrie and M. Sutton,
+        "Area detector based photon correlation in the regime of
+        short data batches: Data reduction for dynamic x-ray
+        scattering," Rev. Sci. Instrum., vol 70, p 3274-3289, 2000.
+
+    """
+    if num_bufs%2 != 0:
+        raise ValueError("number of channels or number of buffers in"
+                         " multiple-taus (must be even)")
+
+    # number of pixels in required roi's, dimensions are : [num_qs]X1
+    num_pixels = np.bincount(q_inds, minlength=(num_qs+1))
+    num_pixels = num_pixels[1:]
+
+    if np.any(num_pixels == 0):
+        raise ValueError("Number of pixels of the required roi's"
+                         " cannot be zero, "
+                         "num_pixels = {0}".format(num_pixels))
+
+    # matrix of auto-correlation function without normalizations
+    G = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                 dtype=np.float64)
+
+    # matrix of past intensity normalizations
+    IAP = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                   dtype=np.float64)
+
+    # matrix of future intensity normalizations
+    IAF = np.zeros(((num_levels + 1)*num_bufs/2, num_qs),
+                   dtype=np.float64)
+
+    # matrix of one-time correlation for required roi's
+    g2 = np.zeros((num_levels, num_qs), dtype=np.float64)
+
+    # correlation for delays, images must be keep for up to maximum
+    # delay in buf
+    buf = np.zeros((num_levels, num_bufs, np.sum(num_pixels)),
+                   dtype=np.float64)
+
+    # to track processing each level
+    cts = np.zeros(num_levels)
+
+    # to increment buffer
+    cur = np.ones((num_levels), dtype=np.int64)
+
+    # to track how many images processed in each level
+    num = np.array(np.zeros(num_levels), dtype=np.int64)
+
+    # starting time for the process
+    t1 = time.time()
+
+    for n in range(0, len(img_stack)):  # changed the number of frames
+
+        cur[0] = (1 + cur[0]) % num_bufs  # increment buffer
+
+        # add image data to the buf to use for correlation
+        buf[0, cur[0] - 1] = (np.ravel(img_stack[n]))[pixel_list]
+
+        # call the _process function for multi-tau level one
+        G, IAP, IAF, num = _process(buf, G, IAP, IAF, q_inds,
+                                    num_bufs, num_pixels, num, level=0,
+                                    buf_no=cur[0] - 1)
+
+        # check whether the number of levels is one, otherwise
+        # continue processing the next level
+        if num_levels > 1:
+            processing = 1
+        else:
+            processing = 0
+
+        # the image data will be saved in buf according to each level then call
+        #  _process function to calculate one time correlation functions
+        level = 1
+        while processing:
+            if cts[level]:
+                prev = 1 + (cur[level - 1] - 1 - 1 + num_bufs)%num_bufs
```

Comment:
and I don't understand why there is the `num_bufs` in the `()` .  If you are taking the mod of it, that is a null operation.